### PR TITLE
use sphinx-design and theme's light/dark image selectors

### LINKED ISSUE #10670: MAINT: Use better CSS for dark mode on website
1. Use https://github.com/pydata/pydata-sphinx-theme/pull/641 where possible
2. Use CSS instead of JS to switch the main logo using https://github.com/pydata/pydata-sphinx-theme/pull/673

Probably a good one for @drammock

### PR DESCRIPTION
**Pull request by drammock**

### CODE REVIEW

**Comment by drammock** [comment]
Circle failure reported upstream: https://github.com/mcmtroffaes/pybtex-docutils/issues/21

**Comment by larsoner** [comment]
In the meantime, it's possible that adding `is ignored and will be removed in Docutils` to our ignores in `conf.py` might work around the issue

**Comment by drammock** [comment]
@mcmtroffaes is usually very quick, let's wait a day and see rather than hastily doing a version-pin or adding a warning-ignore

**Comment by drammock** [comment]
Another failure, reported upstream: https://github.com/numpy/numpydoc/issues/399

seems like using sphinx 5.0 beta (which allows docutils 0.18) is behind these.

File: doc/sphinxext/sphinx_bootstrap_divs/__init__.py [hunk 1]
```diff
@@ -178,7 +123,7 @@ def setup(app):
     except AttributeError:
         app.add_javascript('bootstrap_divs.js')
     app.connect('build-finished', copy_asset_files)
-    for node in (CollapseNode, DetailsNode):
+    for node in (DetailsNode):
```
Comment (larsoner) [inline doc/sphinxext/sphinx_bootstrap_divs/__init__.py:126]:
```suggestion
    for node in (DetailsNode,):
```

**Comment by larsoner** [comment]
```
/home/circleci/project/doc/auto_tutorials/inverse/70_eeg_mri_coords.rst:125: ERROR: Unknown directive type "collapse".
/home/circleci/project/doc/overview/cite.rst:16: ERROR: Error in "dropdown" directive:
unknown option: "class".
/home/circleci/project/doc/overview/cite.rst:35: ERROR: Error in "dropdown" directive:
unknown option: "class".
```

**Comment by hoechenberger** [comment]
@drammock can you please ping me once it builds? I can then have a look and will review :) thanks!

File: doc/install/check_installation.rst [hunk 2]
```diff
@@ -40,8 +40,8 @@ MNE-Python and its dependencies. Typical output looks like this::
     qtpy:          2.0.1 {PySide6=6.2.4}
 
 
-.. collapse:: |hand-paper| If you get an error...
-    :class: danger
+.. dropdown:: If you get an error...
+    :icon: alert-fill
```
Comment (drammock) [inline doc/install/check_installation.rst:44]: here (and in other dropdowns) we can also later add things like `:color: warning` or `:color: danger`, but at present they don't work yet; see https://github.com/executablebooks/sphinx-design/issues/70

File: doc/install/manual_install.rst [hunk 3]
```diff
@@ -24,10 +24,43 @@ instead, a ``conda`` replacement that is **much** faster.
...
+
+    .. tab-item:: Linux, macOS (Intel), Windows
+
+        .. code-block:: console
+
+            $ conda install --channel=conda-forge --name=base mamba
+            $ mamba create --override-channels --channel=conda-forge --name=mne mne
+
+
+    .. tab-item:: macOS (M1 / Apple Silicon)
```
Comment (hoechenberger) [inline doc/install/manual_install.rst:43]: This entire bit is already gone in `main` and has been backported

File: doc/install/index.rst [hunk 4]
```diff
@@ -4,8 +4,64 @@
...
+        +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+
+        .. button-ref:: installers
+            :ref-type: ref
+            :color: primary
+            :shadow:
+            :class: font-weight-bold
+
+            |cloud-download-alt| |ensp| Download Installers
```
Comment (drammock) [inline doc/install/index.rst:35]: This (the `cloud-download-alt`) currently doesn't render correctly, probably should just get removed for now. Upstream issue: https://github.com/executablebooks/sphinx-design/issues/66

**Comment by larsoner** [comment]
Something is wrong in https://output.circle-artifacts.com/output/job/31d5cf6e-0351-41c2-9738-bceed9164c77/artifacts/0/dev/install/index.html

**Comment by drammock** [comment]
> Something is wrong in https://output.circle-artifacts.com/output/job/31d5cf6e-0351-41c2-9738-bceed9164c77/artifacts/0/dev/install/index.html

yep, I noted that in my review.

File: doc/install/index.rst (hunk 4 above)
Comment (larsoner) [reply doc/install/index.rst:35]: Never mind my comments below, then! Feel free to comment out the icons for now then

File: doc/sphinxext/sphinx_bootstrap_divs/__init__.py [hunk 5]
```diff
@@ -66,60 +66,6 @@ def _assemble(node, directive):
...
-        self.assert_has_content()
-        title_text = _(self.arguments[0])
-        extra = _(' show' if 'open' in self.options else '')
-        class_ = {'class': self.options.get('class', 'default')}
-        id_ = nodes.make_id(title_text)
-        node = CollapseNode(title=title_text, id_=id_, extra=extra, **class_)
-        _assemble(node, self)
-        return [node]
-
-
```
[the remark this answers is not in the document]
Comment (larsoner) [reply doc/sphinxext/sphinx_bootstrap_divs/__init__.py:122]: Yes, feel free!

**Comment by hoechenberger** [comment]
I will take a look tomorrow!

In any case we should probably not merge until we have native Apple Silicon installers ready, no?

I believe will get there until Friday.

**Comment by larsoner** [comment]
I think it's easiest to merge this, then deal with M1 whenever it's ready

**Review by hoechenberger — APPROVED** [review]

**Comment by hoechenberger** [comment]
The Circle error is related to some setup problem. I successfully built with my changes locally. Going ahead and merging this one.

**Comment by hoechenberger** [comment]
Thanks @drammock!!

**Pull request merged**